house_interiors_1.logic

```diff
@@ -1,14 +1,70 @@
-area HouseInteriors1
-  room HearthBackDoor
-  room HearthBottom
-  room HearthCheap
-  room HearthDeluxe
-  room HearthGood
-  room HearthOracles
-  room HearthTop
-  room LibraryBottom
-  room LibraryTop
-  room Mayor
-  room PostOffice
-  room SchoolLeft
-  room SchoolRight
+area HouseInteriors1:
+  room HearthBackDoor:
+    node Main:
+      chest HP
+      door Door
+  room HearthBottom:
+    node Main:
+      door South
+      door Left
+      door Middle
+      door Right
+      door Stair
+  room HearthCheap:
+    node Main:
+      door South
+      chest Chest
+  room HearthDeluxe:
+    node Main:
+      door South
+      chest Chest
+  room HearthGood:
+    node Main:
+      door South
+      chest Chest
+  room HearthOracles:
+    node Main:
+      door East
+      door South
+  room HearthTop:
+    node Main:
+      door Stairs
+      door West
+      entrance South
+    node South:
+      exit South
+    Main -> South: item Lantern
+  room LibraryBottom:
+    node Main:
+      door North
+      door Stair
+  room LibraryTop:
+    node Main:
+      door North
+      door South
+    node MainMinish:
+      door Bookcase
+      door South
+  room Mayor:
+    node Main:
+      door East
+      door South
+    node MinishMain:
+      door MinishHouse
+    Main <-> MinishMain: (item Boots & item Cane)
```

Comment:
How does this behave if you come out of the minish home without having entered this way (e.g. by using debug tools)? (You can't use boots as minish, so there should be no edge from MinishMain to Main involving them)

Reply:
I tested, you are simply unable to grow, but how would we write logic for that?

Reply:
Just what you did - if you had to have shrunk here to grow again, there's no need to express being able to grow here at all in the logic (it doesn't care about finding one path through the seed, only a path to each thing that you need to do from the root)